dashboard/test/models/user_test.rb

```diff
@@ -1616,6 +1616,38 @@ def track_progress(user_id, script_level, result, pairings: nil)
     assert_equal [], teacher.reload.permissions
   end
 
+  test 'grant admin permission logs to infrasecurity' do
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Granting UserPermission: environment: #{rack_env}, "\
+        "user ID: #{@teacher.id}, "\
+        "email: #{@teacher.email}, "\
+        "permission: ADMIN",
+        color: 'red'
+      ).
+      returns(true)
+
+    @teacher.admin = true
+    @teacher.save
+  end
+
+  test 'revoke admin permission logs to infrasecurity' do
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Revoking UserPermission: environment: #{rack_env}, "\
+        "user ID: #{@admin.id}, "\
+        "email: #{@admin.email}, "\
+        "permission: ADMIN",
+        color: 'red'
+      ).
+      returns(true)
+
+    @admin.admin = nil
+    @admin.save
+  end
```

Comment:
These tests will fail when run locally on dev environments (as the method early exits based on environment)? Perhaps you want to stub :rack_env?